src/compound_model/CompoundModelFactory.py

```diff
@@ -0,0 +1,403 @@
+
+        Raises:
+            AssertionError: When the number of latent codes are below 1.
+            AssertionError: When shape of the `latent_codes` is not valid.
+            AssertionError: When a attribute is not supported by the controller.
+        """
+        # Check input
+        assert len(latent_codes.shape) > 1
+        num_samples = latent_codes.shape[0]
+        assert num_samples > 0
+        for attr in input.keys():
+            assert attr in self._controller.get_attributes()
+
+        pop = Population(pop_name)
+
+        # Parse input
+        parsed_input = self._controller.parse_native_input(input)
+
+        # Generate images
+        uris, manipulated_latent_codes = self._controller.generate_native(
+            latent_codes, parsed_input
+        )
+
+        # Add images to population
+        pop.add_all(
+            manipulated_latent_codes,
+            latent_codes,
+            uris,
+            [1] * len(uris),
+            append=append,
+            **input,
+        )
+
+        # Clear compound metrics
+        self._clear_compound_metrics(pop_name)
+
+    def generate_population(
+        self, num_samples: int, num_attrs=-1, replace=False
+    ) -> None:
+        """
+        Generates a random population with specified number of samples, or appends samples to a
+        already existing population.
+
+        The number of attributes manipulated for each sample is specified by number of attributes.
+        (Input sampling to be used by the controller is defined by the controller function `sample_random_input()`.)
+
+        An example how the generated population interacts with pre-existing population:
+
+        Let P be a population with 5 number of samples with id 1,2,3,4,5.
+        'generate_population(num_samples = 15, replace=False)' will result in a population with the prexisting samples
+        1-5 and newly created samples 6-15, if replace would be set to true, id 1-5 would be replaced by new samples.
+
+        Another example:
+
+        Let P be a population with 5 number of samples with id 1,2,3,4,5.
+        'generate_population(num_samples = 3, replace=True)' will result in id 1-3 being replaced by new samples and
+        4-5 staying as the were.
+
+        In the last example, if the wanted result would be a new population with just 3 samples, then one would
+        have to manually delete the old population folder first.
+
+        Note that this function clears local and internal compound metrics stored by the compound metric manager for this
+        population.
+
+        Args:
+            num_samples (int): Aspiring number of samples to generate. The actual number of samples to generate
+                depends on `replace`. When `replace` is True, `num_samples` samples will be generated, these samples
+                will replace any pre-existing samples (with that UD) in the population associated this factory. When `replace` is False, generated samples
```

Comment:
"with that UD"? "ID"? maybe break line earlier as well

Reply:
Ye ID